### PR TITLE
[CI] Updates Buildkite configuration

### PR DESCRIPTION
**Pull request by picandocodigo**
Updates configuration to run integration tests on Buildkite.

### CODE REVIEW

**Review by JoshMock — DISMISSED** [review]
looks great! I had it on my "as I have time" list to start moving Python over to Buildkite, but you beat me to it.

File: .buildkite/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+# Default UID/GID to 1000
+# it can be overridden at build time
+ARG BUILDER_UID=1000
+ARG BUILDER_GID=1000
+ENV BUILDER_USER elastic
+ENV BUILDER_GROUP elastic
+ENV PATH="${PATH}:/var/lib/elastic/.local/bin"
+
+# Create user
+RUN groupadd --system -g ${BUILDER_GID} ${BUILDER_GROUP} \
```
Comment (JoshMock) [inline .buildkite/Dockerfile:13]: I think we established that we don't need to do group/user ID stuff in these containers any more, as that was a requirement for DRA, and Python isn't part of DRA any more. @picandocodigo correct me if I'm misremembering.

**Review by JoshMock — APPROVED** [review]
still looks great to me.

**Comment by JoshMock** [comment]
I just double-checked the output of job runs on Buildkite and they're getting pretty similar pass/fail rate as on Jenkins for the integration test suite. Removing unused stuff from `.ci/` is probably the only step left before this is ready to merge.

**Review by pquentin — APPROVED** [review]
Thanks! LGTM. I'm considering merging and and address @JoshMock great comments in a future pull request.

File: .buildkite/Dockerfile (hunk 1 above)
Comment (picandocodigo) [reply .buildkite/Dockerfile:13]: That's right :+1:

**Comment by JoshMock** [comment]
This can also get a catalog-info.yaml file. https://github.com/elastic/catalog-info/pull/563 was merged earlier today.

**Comment by pquentin** [comment]
8.11 was forked after that pull request.

**Pull request merged**